pyproject.toml

```diff
@@ -26,14 +26,14 @@ classifiers = [
 
 [tool.poetry.dependencies]
 python = ">=3.9,<3.11"
-numpy = "1.23.5"
-scipy = "1.8.0"
-numba = "0.56.4"
-thewalrus = "0.19.0"
-rich = "10.15.1"
-matplotlib = "3.5.0"
+numpy = "^1.23.5"
+scipy = "^1.8.0"
+numba = "^0.56.4"
+thewalrus = "~0.19.0"
+rich = "^10.15.1"
+matplotlib = "^3.5.0"
 tqdm = "^4.65.0"
```

Comment:
```suggestion
```

Reply:
I noticed tqdm is actually not used anywhere!

Reply:
let's see how tests are doing first